### ISSUE TITLE
Type hints check of array type does not accept empty array as valid value.

### ISSUE THREAD

**Issue by discordier** [scrutinizer-ci/php-analyzer#543]
When an interface hints to be of array type, an empty array is not considered valid.
```php
interface FooInterface
{
  /**
   * This is foobar.
   *
   * @return string[]|null
   */
  public function bar();
}

class Foo implements FooInterface
{
  /**
   * {@inheritdoc}
   */
  public function bar()
  {
    return array();
  }
}
```

This yields:
```
The return type of return array(); (array) is incompatible with the return type declared by the 
interface FooInterface::searchFor of type string[]|null.
```

IMO the empty array should be accepted as valid return value as even an empty array is what it is... an array. :)

see https://scrutinizer-ci.com/g/MetaModels/core/inspections/9c54d742-b686-4068-8153-3047087ffae2/issues/files/src/MetaModels/Attribute/Base.php?status=new&orderField=path&order=asc&issueId=60f88d44-76c3-4b5f-8026-4d7575032ad5

**Comment by tristanlins** [comment]
+1

**Comment by discordier** [comment]
It might be that #539 plays into the same part somehow as there scrutinizer suggests to change `int[]` to `array`.
